resources/source.rb

```diff
@@ -0,0 +1,65 @@
+#
+# Author:: Blair Hamilton (bhamilton@draftkings.com>)
+# Author:: Jonathan Morley (morley.jonathan@gmail.com>)
+# Cookbook Name:: chocolatey
+# Resource:: source
+#
+# Copyright 2015, Blair Hamilton
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+require 'chef/mixin/shell_out'
+include Chef::Mixin::ShellOut
+
+include ::Chocolatey::Helpers
+
+property :name, kind_of: String, name_attribute: true
+property :source, kind_of: String
+
+default_action :add
+
+load_current_value do
+  require 'nokogiri'
```

Comment:
nokogiri really needs to be replaced by rexml here.  nokogiri is endless pain and should only be used if performance is an issue or some edge case is hit with rexml.  rexml is built-in and avoids all the issues surrounding building and installing nokogiri.

Reply:
There is an implementation in the Puppet provider that uses rexml.